examples/island/Island.cpp

```diff
@@ -123,14 +123,16 @@ int main()
     {
         statusText.setString("Shaders and/or Vertex Buffers Unsupported");
     }
-    else if (!terrainShader.loadFromFile("resources/terrain.vert", "resources/terrain.frag"))
+    else if (!(terrainShader = sf::Shader::loadFromFile("resources/terrain.vert", "resources/terrain.frag")))
     {
         prerequisitesSupported = false;
 
         statusText.setString("Failed to load shader program");
     }
     else
     {
+        terrainStates = sf::RenderStates(&*terrainShader);
```

Comment:
Don't keep `terrainStates` as a data member, just create it on the fly around line 243.